### PR TITLE
Prateek school api

### PR DESCRIPTION
**Pull request by prateekj01**

### CODE REVIEW

File: Prateek/postgresassignment/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "trying",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
```
Comment (prateekj01) [inline Prateek/postgresassignment/package.json:5]: change the index.js file

File: Prateek/postgresassignment/package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "scripts": {
+    "build": "tsc -p .",
+    "lint": "eslint ./",
+    "test": "echo \"Error: no test specified\" && exit 1"
```
Comment (prateekj01) [inline Prateek/postgresassignment/package.json:9]: add start

File: Prateek/postgresassignment/src/controller/class.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  try {
+    var id: number = parseInt(ctx.params.id);
+    if (id === undefined || typeof id !== "number") {
+      ctx.response.status = 400;
+      ctx.response.type = "text/html";
+      ctx.body = "Bad Request";
+      return;
+    }
+    let [rows]: Array<{ rows: student_det }> = [];
+    rows = await serviceclass.get_student_by_standard(id);
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/controller/class.ts:51]: should be inside try

File: Prateek/postgresassignment/src/controller/result.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      req.standard === undefined ||
+      req.marks === undefined
+    ) {
+      ctx.response.status = 400;
+      ctx.response.type = "text/html";
+      ctx.body = "Bad Request";
+      return;
+    }
+
+    if (
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/controller/result.ts]: Use JOI for validation

File: Prateek/postgresassignment/src/controller/subject.ts [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+  standard: number;
+}
+
+interface stud_marks {
+  subcode: number;
+  subject: string;
+  marks: number;
+}
+
+export async function getSubject(ctx: Context) {
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/controller/subject.ts:25]: google koa typescript context

File: Prateek/postgresassignment/src/database/db.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+const { Client } = require("pg");
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/database/db.ts:1]: process.env and use config or environ file

File: Prateek/postgresassignment/src/routes/routes.ts [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+
+app.use(router.routes());
+app.use(router.allowedMethods());
+app.use(async (ctx: any) => {
+  ctx.response.status = 404;
+  ctx.response.type = "text/html";
+  ctx.body = "Not Found";
+});
+
+app.listen(port, () => {
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/routes/routes.ts]: export routes in app and server

File: Prateek/postgresassignment/src/services/class.ts [hunk 8]
```diff
@@ -0,0 +1,21 @@
+import { client as sqlclient } from "../database/db";
+
+export async function get_class() {
+  try {
+    await sqlclient.query("SET search_path TO school");
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/services/class.ts]: remove this and make a default schema to be selected

File: Prateek/postgresassignment/src/services/result.ts [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+  } catch (e) {
+    throw Error(e);
+  }
+}
+
+export async function subject_length(roll_num: number) {
+  try {
+    await sqlclient.query("SET search_path TO school");
+    return await sqlclient.query(
+      `SELECT COUNT(*) FROM (SELECT cs.subcode FROM Class_schedule AS cs,Standards AS s WHERE cs.roll_num=${roll_num} AND cs.Standard=s.standard) AS S`
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/services/result.ts:53]: sub query not required

File: Prateek/postgresassignment/src/services/student.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+export async function get_student_length() {
+  try {
+    await sqlclient.query("SET search_path TO school");
+    return await sqlclient.query("SELECT Count(*) FROM Students");
+  } catch (e) {
+    throw Error(e);
+  }
+}
+
+export async function add_student(
```
Comment (prateekj01) [inline Prateek/postgresassignment/src/services/student.ts:21]: add a return type to the function.

**Review by arjungowda-kr — APPROVED** [review]

**Pull request merged**